bezierDrawing.py

```diff
@@ -0,0 +1,27 @@
+import inkex
+from inkex import Rectangle, Transform
+from inkex import Vector2d
+from inkex import Circle, Ellipse, Polygon, Line, Rectangle, Path
+
+
+class ExtendedHatchFill(inkex.EffectExtension):
+    """EffectExtension to fill selected objects red"""
+
+    def add_arguments(self, pars):
+        pars.add_argument("--even_color", type=inkex.Color, default=inkex.Color("blue"))
+
+    def effect(self):
+        s = self.svg
+        a = [['M ', [100, 100]],
+             [' L ', [200, 300]],
+             [' L ', [600, 300]],
+             [' L ', [800, 550]],
+             [' L ', [1000, 300]],
+             [' Z', []], ]
+
+        bz = Path(d=a)  # or  bz = Path(d = f'{a}')  or bz = Path(f'{a}')
```

Comment:
bz = PathElement.new(a)
As you can see from Inkscape sources "Path" is a subclass of "list". In order to add it to the document we have to use ShapeElement descendants.

Reply:
or ```
p = Path(a)
bz = PathElement.new(p)
```
in case you need the path for further modifications.

Reply:
@[toxa23](https://github.com/toxa23) thanks, it worked
"```s = self.svg
        a = [['M', [200, 300]],
            ['Q', [400, 50, 600, 300]],
            ['T', [1000, 300]],
            ['Z', []],]
        bz = PathElement.new(a)
        bz.style = { "fill": "none", "stroke": "black" }```"

Reply:
Or is it better like this



```def effect(self):
        s = self.svg
        bz = PathElement.new("M 100 100 L 300 100 L 200 300 z")
        bz.style = { "fill": "red", "stroke": "black" }
        s.add(bz)

Reply:
Pycharm rejects my commits, so I will write here.

```
    def effect(self):
        s = self.svg
        ds = 30
        pic = ImageGrab.grab()
        #hatching bezier selected rectangular objects
        for elem in self.svg.selection:
            b = elem.bounding_box()
            dxx = b.left # object offsets for bitmap hatching
            dyy = b.top
            for y in range(int(b.top + ds/2), int(b.top + b.height- ds/2), int(ds / 2)):
                for x in range(int(b.left), int(b.left + b.width - ds / 2 ), ds):
                    dy = randint(10,30)
                    dx = randint(30,50)
                    a = [['M', [x, y]],
                        ['Q', [x + dx, y + dy, x + 40, y]],
                        ['T', [x + 40, y]]]

                    # below is code for bitmap hatching objects (in the example by red)
                    # without debugger I can't understand why it gives me an error
                    # The error is shown at line 23 (for ...)

                    # r, g, b = pic.getpixel((int(x), int(y)))
                    # if r == 255 and g == 0 and b == 0:
                    #     hatch = Ellipse(cx=f'{x - dxx}', cy=f'{y - dyy}', rx=f'{10}', ry=f'{10}')
                    #     hatch.set_id(f'hatch' + '{x}' + '{y}')
                    #     hatch.style = { "fill": "green", "stroke": "black" }
                    #     s.add(hatch)

                    bz = PathElement.new(a)
                    bz.style = { "fill": "none", "stroke": "black"}
                    s.add(bz)

Reply:
![image](https://user-images.githubusercontent.com/15789199/170844207-71853e8c-aca2-4322-99bb-6ce61fb99fcd.png)

Reply:
oops ... b is the color (from RGB) and b is the bounding_box